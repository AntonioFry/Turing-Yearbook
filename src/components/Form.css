.form {
  background-color: rgba(255, 255, 255, 0.95);
  display: flex;
  flex-direction: column;
  padding: 10px;
  align-items: center;
  color: limegreen;
  width: 100%;
}

select,
button,
input {
  margin: 5px;
  width: 200px
}