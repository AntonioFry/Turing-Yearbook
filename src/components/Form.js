import React, { Component } from "react";
import './Form.css';

class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      quote: '',
      superlative: ''
    }
  }

  handleChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }

  render() {
    return (
      <form>
        <input
          type='text'
          placeholder='name'
          name='name'
          value={this.state.name}
          onChange={event => this.handleChange(event)}
        />
        <input
          type='text'
          placeholder='quote'
          name='quote'
          value={this.state.quote}
          onChange={event => this.handleChange(event)}
        />
        <input
          type='text'
          placeholder='superlative'
          name='superlative'
          value={this.state.superlative}
          onChange={event => this.handleChange(event)}
        />
      </form>
    )
  }
}

export default Form;