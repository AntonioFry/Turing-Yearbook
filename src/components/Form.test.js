import React from 'react';
import { shallow, mount } from 'enzyme';
import Form from './Form';
import { exportAllDeclaration } from '@babel/types';

describe('Form', () => {

  var wrapper;
  var mockFunction;
  beforeEach(function() {
    mockFunction = jest.fn();
    wrapper = shallow(<Form addPerson={mockFunction} />)
  });

  it('match the snapshot with all the data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke addPerson when button is clicked', () => {

    wrapper.find(".add-person").simulate('click');

  });
})