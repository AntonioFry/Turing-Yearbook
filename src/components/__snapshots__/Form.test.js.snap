// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form match the snapshot with all the data passed in correctly 1`] = `
<form
  className="form"
>
  <input
    name="name"
    onChange={[Function]}
    placeholder="name"
    type="text"
    value=""
  />
  <input
    name="quote"
    onChange={[Function]}
    placeholder="quote"
    type="text"
    value=""
  />
  <input
    name="superlative"
    onChange={[Function]}
    placeholder="superlative"
    type="text"
    value=""
  />
  <select
    name="category"
    onChange={[Function]}
  >
    <option
      name="staff"
      value="staff"
    >
      Staff
    </option>
    <option
      name="student"
      value="students"
    >
      Students
    </option>
  </select>
  <button
    className="add-person"
    onClick={[Function]}
  >
    Submit
  </button>
</form>
`;
